src/server.rs

```diff
@@ -33,20 +41,64 @@ pub struct ServerState {
 /// Run the orchestrator server
 pub async fn run(
     http_addr: SocketAddr,
-    _tls_cert_path: Option<PathBuf>,
-    _tls_key_path: Option<PathBuf>,
-    _tls_client_ca_cert_path: Option<PathBuf>,
+    health_http_addr: SocketAddr,
+    tls_cert_path: Option<PathBuf>,
+    tls_key_path: Option<PathBuf>,
+    tls_client_ca_cert_path: Option<PathBuf>,
     config_path: PathBuf,
 ) -> Result<(), Error> {
     let config = OrchestratorConfig::load(config_path).await;
     let orchestrator = Orchestrator::new(config).await?;
     let shared_state = Arc::new(ServerState { orchestrator });
 
-    // TODO: configure server TLS
+    // Separate HTTP health server without TLS for probes
```

Comment:
Can you divide all the different functionalities in this function into logic steps and document them here. So sort of like:
```
1. Configure handlers with routes
2. Configure TLS if requested with each of the routes
...
```
and then in your comments just above those implementations, you can also associated same step number. That way it would be a bit more easy to follow